src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java

```diff
@@ -437,6 +447,24 @@ public void execute()
 
             List<ResourceTransformer> resourceTransformers = getResourceTransformers();
 
+            if ( createDependencyReducedPom )
```

Comment:
You have moved this before the Shader works. No changes in behavior?

Reply:
With the tests present in the project today 2 fail with my changes in place.
So this does indeed need some rework.

Reply:
Apparently the problem that failed the tests made two files. 
Fix that now all tests pass.

Still the question "No changes in behavior?" remains a valid one.

For now: Let's assume there IS a change the current tests did not pick up.
I'll look into it.

Reply:
My verification steps:

0. All integration tests pass.
1. Checkout both master and my version into separate directories.
2. Run in both: `MAVEN_OPTS=-Dinvoker.parallelThreads=2 mvn clean verify -Prun-its` under javac 11.0.11
3. Get this tool to compare the jars https://github.com/scala/jardiff
4. Run these magical incantations on my Linux machine:
`find target/it -type f -name '*.jar' | while read JAR ; do java -jar ../jardiff.jar ${JAR} ../maven-shade-plugin-master/${JAR} ; done > jar-changes.log`
`find target/it -type f -name '*pom*' | while read POM ; do diff  ${POM} ../maven-shade-plugin-master/${POM} ; done > pom-changes.log`
5. Manually go through the output.
Differences logged:
- Files like surefirebooter2116138492290729396.jar are unique (Ignore)
- The test files I created are only present in mine (Ignore)
- All files have a difference in the timestamps when the files were generated (pom.properties) (Ignore)
- Some tests include the plugin itself and show differences because of the new feature (Ignore)

@michael-o Is this sufficient verification?

Reply:
Will verify!